src/main.rs

```diff
@@ -21,43 +21,35 @@ use std::io::Error;
 use std::path::Path;
 use std::process;
 
-fn write_mobj_to_file(file: &mut File, moray_obj: Value) -> Result<(), Error> {
-    let manta_obj = match sharkspotter::manta_obj_from_moray_obj(&moray_obj) {
-        Ok(mo) => mo,
-        Err(e) => {
-            eprintln!("{}", e);
-            return Ok(());
-        }
-    };
+fn write_mobj_to_file(
+    file: &mut File,
+    moray_obj: Value,
+    full_object: bool,
+) -> Result<(), Error> {
+    let out_obj: Value;
 
-    let object_id = match manta_obj.get("objectId") {
-        Some(oid) => match serde_json::to_string(oid) {
-            Ok(o) => o,
+    if !full_object {
+        out_obj = match sharkspotter::manta_obj_from_moray_obj(&moray_obj) {
+            Ok(mo) => mo,
             Err(e) => {
-                eprintln!(
-                    "Could not deserialize objectId {:#?}, {}",
-                    manta_obj, e
-                );
+                eprintln!("{}", e);
                 return Ok(());
             }
-        },
-        None => {
-            eprintln!("Could not get objectId from value {:#?}", manta_obj);
-            return Ok(());
         }
-    };
-
-    println!("{}", object_id);
+    } else {
+        out_obj = moray_obj;
+    }
 
-    let buf = serde_json::to_string(&manta_obj)?;
-    file.write_all(buf.as_bytes())?; // TODO: match
+    let buf = serde_json::to_string(&out_obj)?;
+    file.write_all(buf.as_bytes())?;
     file.write_all(b"\n")?;
```

Comment:
So you could remove "File" here and make it more generic to anything that implements write. It would also allow you to skip the to_string() step. Just a nit though, feel free to keep it as is if you want.

```Rust
  fn write_mobj_to_file<W: Write>(
      mut writer: W,
      moray_obj: Value,
      full_object: bool,
  ) -> Result<(), Error> {
      let out_obj: Value;

      if !full_object {
          out_obj = match sharkspotter::manta_obj_from_moray_obj(&moray_obj) {
              Ok(mo) => mo,
              Err(e) => {
                  eprintln!("{}", e);
                  return Ok(());
              }
          }
      } else {
          out_obj = moray_obj;
      }

      serde_json::to_writer(&mut writer, &out_obj)?;
      writer.write_all(b"\n")?;

      Ok(())
  }
```